### PR TITLE
Enabling DAC for STM32 F429ZI Platform

### PR DESCRIPTION
**Pull request by sarthsmart**
Enabled DAC support for STM32F429ZI target platform and verified by connecting the PA4 pin to oscilloscope. Then executed the test application to generate the sawtooth signal pattern using DAC, and the expected sawtooth wave could be seen on the oscilloscope.

### CODE REVIEW

File: dts/arm/st/f4/stm32f4.dtsi [hunk 1]
```diff
@@ -408,6 +408,15 @@
 			#io-channel-cells = <1>;
 		};
 
+		dac1: dac@40007400 {
```
Comment (erwango) [inline dts/arm/st/f4/stm32f4.dtsi:411]: This peripheral is not present on all STM32F4 variants.
Please move this definition in following files:
stm32f405.dtsi, stm32f413.dtsi and stm32f446.dtsi.

**Review by erwango — CHANGES_REQUESTED** [review]

File: dts/arm/st/f4/stm32f4.dtsi (hunk 1 above)
Comment (sarthsmart) [reply dts/arm/st/f4/stm32f4.dtsi:411]: @erwango 
1: We don't have these boards, I can still raise git pull request with additions done for F405, F413 & F446 dtsi files, is it ok?
2: What about F429(ZI)? We have this board.

Comment (erwango) [reply dts/arm/st/f4/stm32f4.dtsi:411]: 1: Please do this inside current PR, this is fine.
2: Please have a look to inclusion scheme (429 includes 427, 427 includes ...)

Comment (sarthsmart) [reply dts/arm/st/f4/stm32f4.dtsi:411]: > 1: Please do this inside current PR, this is fine. => Added definition inside F405, F413 & F446 dtsi files. For F405, do we need to add new platform under boards/arm hierarchy?
For F413, the platform is nucleo_f413zh & for F446, the platform is nucleo_f446re. Please confirm.

> 2: Please have a look to inclusion scheme (429 includes 427, 427 includes ...) => Updated the dtsi code within inclusion.

Comment (erwango) [reply dts/arm/st/f4/stm32f4.dtsi:411]: You don't need to enable DAC on those boards. The only point is to correctly enable DAC on .dtsi

Comment (sarthsmart) [reply dts/arm/st/f4/stm32f4.dtsi]: @erwango Corrected the definition changes for STM32 F429ZI platform.
Enabled DAC code on stm32f405.dtsi, stm32f413.dtsi and stm32f446.dtsi.

**Review by erwango — CHANGES_REQUESTED** [review]
Overall change is fine, but please squash last commit into the first one (we don't merge fix commits)

**Comment by sarthsmart** [comment]
> Overall change is fine, but please squash last commit into the first one (we don't merge fix commits)

Checking is done for the commit. Thanks Erwan.

**Review by erwango — APPROVED** [review]

**Review by martinjaeger — APPROVED** [review]
The first commit saying `dts/arm: STM32: enable dac support for stm32f4 in dtsi` is not complete, as it does not add the DAC to all F4 series MCUs where it is available (see [here](https://www.st.com/content/st_com/en/products/microcontrollers-microprocessors/stm32-32-bit-arm-cortex-mcus/stm32-high-performance-mcus/stm32f4-series.html#products)).

I would appreciate to make it complete, but I won't insist.

**Pull request merged**